### DISCUSSION TITLE
Using `GITHUB_TOKEN` to setup auth for extensions

### DISCUSSION THREAD

**Discussion by kohlivarun5** [coder/code-server#6477] [Q&A]
I see that there is support for `GITHUB_TOKEN` to be passed in via the env, and tests [here](https://github.com/coder/code-server/blob/80dfb2b2a9ef36c527e12d2aa0ecf80edfe928fa/test/e2e/github.test.ts#L4) seem to suggest that with that, user should not need to log-in again.

In my environment, when I try this, the extension asks to sign-in and routes to Github for auth 
<img width="552" alt="Screenshot 2023-10-06 at 2 09 49 PM" src="https://github.com/coder/code-server/assets/896953/04956be0-3cc2-4930-bf2a-57516a285f64">

1) Anything that I might be missing here ? I should mention, I am trying this against an enterprise instance, by setting `"github-enterprise.uri"` in code-server settings.json
2) Is there a way to make it even more seamless, so that the user does not have to click `Sign-In` & `Allow` ? Ideally, we would be able to set it up such that the extension just works as pre-authorized 

#### Info about code-server

```
code-server: v4.17.1
Code: 1.82.2
Commit: 2eba7af117ea58d45a6c6449ee4fe63c8d4d53aa
Date: 2023-09-29T21:59:34.461Z
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Safari/605.1.15
```

**Comment by kohlivarun5** [comment]
@code-asher Is this working as expected, or might I be missing some setup ?

**Reply by code-asher** [reply]
It should work with the GitHub pull requests extension at least, but I am not sure about others.  Is that the one you are using or is it a different extension?  Could also be worth testing it with desktop VS Code (launch with `--github-auth <token>` I think).

I do not believe there is a way to make it work without the user explicitly allowing.  Probably this would be considered a security issue if extensions could access credentials without asking.

**Reply by code-asher** [reply]
Hmm or maybe it is related to the enterprise URL.  That is not something I have tested.  If it works in desktop VS Code it should work in code-server though.

**Reply by kohlivarun5** [reply]
Thanks for the pointer, will test desktop VSCode to confirm behavior

**Reply by benz0li** [reply]
@code-asher This does not work as stated. Instead of

> // It should ask to select an account, one of which will be the one we
> // pre-injected.

one has to sign in to GitHub:

<img width="330" alt="sign-in-to-github" src="https://github.com/coder/code-server/assets/20233085/6ffbd7f9-6749-4c09-8b67-f3c54403db80">

```bash
$ code-server --version
4.18.0 d7a2b4936af1bfd80cb96f2567af68badc2325e3 with Code 1.83.1
```

---

I found out, that [compared to Codespaces] environment variable `GITHUB_TOKEN` is unset in the code-server Terminal – even though being set before starting code-server.
👉 Environment variable `CODE_SERVER_VERSION` – even though being set in the Dockerfiles of my JupyterLab language docker-stacks – is also unset in the code-server Terminal.

**Reply by benz0li** [reply]
Regular usage of the 'GitHub Pull Requests and Issues' extensions, i.e. after properly signing into GiHub, seems to work fine since code-server v4.17.1.  
ℹ️ Until the cache (website data) of the browser is cleared, of course.

code-server v4.16.1 (and maybe earlier versions) behaved slightly different: When setting environment variable `GITHUB_TOKEN`, 'Accounts' (Activity Bar) showed you as signed into GitHub as soon as a folder containing a GitHub repository was opened.
👉 This is not the case anymore.

For me, being more of a GitLab user, all this is *nice to have* and not *mandatory*.

**Reply by code-asher** [reply]
Ahh OK thank you for the report.  It might be a bug with VS Code's `--github-auth`, all we do now is set that to the value in the environment variable or flag, but we will need to investigate further.
